index.js

```diff
@@ -1,199 +1,277 @@
             }
         }
     }
 
     _ensureLoggedIn() {
-        return this.rp.get(`${this.controller.href}api/${this.opts.unifios ? 'users/' : ''}self`)
-            .catch(() => {
-                return this._login();
-            });
+        return new Promise((resolve, reject) => {
+            this.instance.get(`${this.controller.href}api/${this.unifios ? 'users/' : ''}self`).then(() => {
+                resolve(true)
+            }).catch(() => {
+                this._login().then(() => {
+                    resolve(true)
+                }).catch(error => {
+                    reject(error)
+                })
+            })
+        })
     }
 
     _url(path) {
-        if (this.opts.unifios) {
-            // unifios using an proxy, set extra path
+        if (this.unifios) {
             if (path.indexOf('/') === 0) {
-                return `${this.controller.href}proxy/network/${path}`;
+                return `${this.controller.href}proxy/network/${path}`
             }
-            return `${this.controller.href}proxy/network/api/s/${this.opts.site}/${path}`;
+            return `${this.controller.href}proxy/network/api/s/${this.opts.site}/${path}`
         }
         else {
             if (path.indexOf('/') === 0) {
-                return `${this.controller.href}${path}`;
+                return `${this.controller.href}${path}`
             }
-            return `${this.controller.href}api/s/${this.opts.site}/${path}`;
+            return `${this.controller.href}api/s/${this.opts.site}/${path}`
         }
     }
 
     get(path) {
-        return this._ensureLoggedIn()
-            .then(() => {
-                return this.rp.get(this._url(path));
-            });
+        return new Promise((resolve, reject) => {
+            this._ensureLoggedIn().then(() => {
+                this.instance.get(this._url(path)).then(response => {
+                    resolve(response.data)
+                }).catch(error => {
+                    reject(error)
+                })
+            }).catch(error => {
+                reject(error)
+            })
+        })
     }
 
     del(path) {
-        return this._ensureLoggedIn()
-            .then(() => {
-                return this.rp.del(this._url(path));
-            });
+        return new Promise((resolve, reject) => {
+            this._ensureLoggedIn().then(() => {
+                this.instance.del(this._url(path)).then(response => {
+                    resolve(response.data)
+                }).catch(error => {
+                    reject(error)
+                })
+            }).catch(error => {
+                reject(error)
+            })
+        })
     }
 
     post(path, body) {
-        return this._ensureLoggedIn()
-            .then(() => {
-                return this.rp.post(this._url(path), {body});
-            });
+        return new Promise((resolve, reject) => {
```

Comment:
Why create new promises like this? the `_ensureLoggedIn` method and axios returns promises already